x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/BlobAnalyzeAction.java

```diff
@@ -368,20 +368,7 @@ private void readOnNodes(List<DiscoveryNode> nodes, boolean beforeWriteComplete)
                         new ActionListenerResponseHandler<>(new ActionListener<>() {
                             @Override
                             public void onResponse(GetBlobChecksumAction.Response response) {
-                                if (beforeWriteComplete == false && response.isNotFound()) {
-                                    readNodesListener.onFailure(
-                                        new RepositoryVerificationException(
-                                            request.getRepositoryName(),
-                                            "["
-                                                + blobChecksumRequest
-                                                + "] (after write complete) failed on node ["
-                                                + node
-                                                + "]: blob not found"
-                                        )
-                                    );
-                                } else {
-                                    readNodesListener.onResponse(makeNodeResponse(node, beforeWriteComplete, response));
-                                }
+                                readNodesListener.onResponse(makeNodeResponse(node, beforeWriteComplete, response));
```

Comment:
Note that we don't completely ignore the not-found case, we handle it correctly in `onReadsComplete()`:

https://github.com/elastic/elasticsearch/blob/f2c18d4d580921d7370b5c12fd9e5b053ba25c26/x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/BlobAnalyzeAction.java#L473-L483

I added this extra check here during some testing and clearly forgot to remove it.